In [1]:
import pandas as pd
from ydata_profiling import ProfileReport
import sweetviz as sv

# Exploración de datos generando profit

In [5]:
df = pd.read_csv("Prueba de modelamiento para Cientifico de Datos 2023.csv", encoding='latin-1', sep=';')

In [6]:
df

loteNo departamento_anonym empresa_anonym  \
0     5e4604c71c555bfa922bd644                dp_0           em_0   
1     5e4604c71c555bfa922bd68c                dp_0           em_0   
2     5e4604c71c555bfa922bd6b5                dp_0           em_0   
3     5ea0d184cbe8d10cee382f9b                dp_0           em_0   
4     5ef2216bde8636c49c3b8530                dp_0           em_0   
...                        ...                 ...            ...   
2520  6453f7e618bf3e15193e673b                dp_6          em_71   
2521  6455212918bf3e15197d4918                dp_3          em_63   
2522  6459810518bf3e1519b1d811                dp_0          em_75   
2523  645a406018bf3e1519de455c                dp_0          em_29   
2524  645a406018bf3e1519de4565                dp_0          em_29   

     granja_anonym proveedorCorregido_anonym fabricaAlimento genetica  \
0             gr_0                      pn_0            pp_0     lg_0   
1             gr_0                      pn_0            pp_0     lg_0   
2             gr_0                      pn_0            pp_0     lg_0   
3             gr_0                      pn_0            pp_0     lg_0   
4             gr_0                      pn_0            pp_0     lg_0   
...            ...                       ...             ...      ...   
2520         gr_81                      pn_0            pp_2     lg_0   
2521         gr_69                      pn_6            pp_8     lg_0   
2522         gr_85                      pn_0            pp_0     lg_0   
2523         gr_30                      pn_0            pp_0     lg_0   
2524         gr_30                      pn_0            pp_0     lg_0   

      duracionCeba  edadFinal  edadInicial  ...  pesoFinal  fechaSalida  \
0            72.00     152.00        80.00  ...   103.5600    9/09/2019   
1            81.00     154.00        73.00  ...   100.6400   23/10/2019   
2            79.00     152.00        73.00  ...   102.0700   19/11/2019   
3            76.00     155.00        79.00  ...   114.1200   13/03/2020   
4            74.00     154.00        80.00  ...   118.6400    1/04/2020   
...            ...        ...          ...  ...        ...          ...   
2520         60.00     146.53        86.53  ...    95.9700   30/03/2023   
2521         93.19     165.16        71.97  ...   122.3664    1/03/2023   
2522         83.00     155.09        72.09  ...    95.8500   31/03/2023   
2523         72.00     140.29        68.29  ...    84.5100    6/08/2022   
2524         69.00     139.35        70.35  ...    97.1400   17/03/2023   

     anioSalida  semesreSalida  anioSemestreSalida asesor numFinalAnimales  \
0          2019              2             2019-S2   rt_0              220   
1          2019              2             2019-S2   rt_0              187   
2          2019              2             2019-S2   rt_0              194   
3          2020              1             2020-S1   rt_0              203   
4          2020              1             2020-S1   rt_0              185   
...         ...            ...                 ...    ...              ...   
2520       2023              1             2023-S1  rt_12             1298   
2521       2023              1             2023-S1  rt_18              645   
2522       2023              1             2023-S1   rt_0              238   
2523       2022              2             2022-S2   rt_0             1007   
2524       2023              1             2023-S1   rt_0             1092   

      numInicialAnimales  mortalidad  conversionAlimenticia  
0                    226        2.65                 2.1760  
1                    201        6.97                 2.2090  
2                    203        4.43                 2.1350  
3                    208        2.40                 2.1050  
4                    189        2.12                 2.2220  
...                  ...         ...                    ...  
2520                1338        2.99          

In [7]:
df["loteNo"].duplicated().sum()

0

In [8]:
# Crear el informe
profile = ProfileReport(df, title="Análisis del DataFrame", explorative=True)
profile.to_file("reporte.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

In [9]:


# Crear el informe
report = sv.analyze(df)

                                             |          | [  0%]   00:00 -> (? left)

In [10]:
report.show_html("reporte_sweetviz.html")

Report reporte_sweetviz.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.


In [11]:
df.columns

Index(['loteNo', 'departamento_anonym', 'empresa_anonym', 'granja_anonym',
       'proveedorCorregido_anonym', 'fabricaAlimento', 'genetica',
       'duracionCeba', 'edadFinal', 'edadInicial', 'pesoInicial', 'pesoFinal',
       'fechaSalida', 'anioSalida', 'semesreSalida', 'anioSemestreSalida',
       'asesor', 'numFinalAnimales', 'numInicialAnimales', 'mortalidad',
       'conversionAlimenticia'],
      dtype='object')

In [12]:
df["departamento_anonym"].value_counts(dropna=False)

departamento_anonym
dp_3    1215
dp_1     453
dp_0     386
dp_2     368
dp_5      74
dp_6      11
dp_7      10
dp_4       8
Name: count, dtype: int64

In [13]:
df["empresa_anonym"].value_counts(dropna=False)

empresa_anonym
em_46    250
em_2     119
em_17    117
em_41    115
em_52    112
        ... 
em_30      6
em_74      6
em_76      6
em_62      5
em_65      4
Name: count, Length: 77, dtype: int64

In [14]:
df["conversionAlimenticia"].value_counts()

conversionAlimenticia
2.1900    20
2.0500    20
2.1300    16
2.2400    16
2.2500    15
          ..
2.1115     1
2.1741     1
2.0526     1
2.2317     1
2.1587     1
Name: count, Length: 1692, dtype: int64

# Explicación de datos a partir de reportes obtenidos

- Mas de la mitad de la muestra de cerdos analizada, pertenece a los departamentos
 dp_3 y dp_1.
- Existen 13 empresas, pero más del 60% de los cerdos pertenecen a 3 empresas: e, m, -.
- Existen 87 granjas distintas, lo cual hace que la población este bien distribuida,
  siendo las granjas 48, 2 y 18 las que más lotes de cerdos distintos posee.}
- Existen 9 fabricas de alimento, donde 5 de ellas contribuyen cada una menos del 10% de la población.
- La genética de los lotes de cerdos principalmente se encuentra en lg_0.
- El periodo de ceba o de engorne en los lotes análizados en general es de 84 semanas.
- Este periodo esta altamente correlacionado con la edad final con la que sale cada lote de cerdos.
- La edad inicial, aparenta una distribucción bimodal con modos en 72 y 79 semanas  y como es de esperar esta variable esta altamente correlacionada con el peso inicial.
- Algo interesante es como cambia la distribución de pesos en el tiempo (peso inicial a final). Iniciamos con una distribución centrada en 31 kg y concluimos con una distribución con media en 113K e inclinada hacia la derecha, lo cual es bueno, por que se tiene al final una mayor diferencia entre de pesos, es decir una ganancia mayor de peso que provoca una conversión menor.
- El periodo de salida es de el primer semestre del 2019 y el segundo del 2023.
- Se tiene 26 acesores, siendo el más solicitado el tr_18.
- La mortalidad media es de 1.8% pero tenemos extemos superiores (valores superiores al percentil 95) del 5%.
- Tenemos nuestra variable de interés que es la conversión con un valor medio de 2.23, es decir, por cada 2.23 kg que come el cerdo, este incrementa en 1 su peso.
- La Razón de Correlación mide la fuerza de asociación entre una variable categórica y una variable continua, nos muestra que la granja y la empresa y el asesor, fabrica de alimento están altamente correlacionados.

In [19]:
cols = [
     'fabricaAlimento',
    'granja_anonym', 
    'genetica',
    'pesoInicial',
    'pesoFinal',
    'semesreSalida',
    'asesor',
    'mortalidad',
    'conversionAlimenticia'
]

In [20]:
df[cols]

fabricaAlimento granja_anonym genetica  pesoInicial  pesoFinal  \
0               pp_0          gr_0     lg_0      30.9500   103.5600   
1               pp_0          gr_0     lg_0      26.6100   100.6400   
2               pp_0          gr_0     lg_0      28.3200   102.0700   
3               pp_0          gr_0     lg_0      29.3400   114.1200   
4               pp_0          gr_0     lg_0      34.0400   118.6400   
...              ...           ...      ...          ...        ...   
2520            pp_2         gr_81     lg_0      37.6400    95.9700   
2521            pp_8         gr_69     lg_0      32.6621   122.3664   
2522            pp_0         gr_85     lg_0      27.7200    95.8500   
2523            pp_0         gr_30     lg_0      28.0400    84.5100   
2524            pp_0         gr_30     lg_0      29.0000    97.1400   

      semesreSalida asesor  mortalidad  conversionAlimenticia  
0                 2   rt_0        2.65                 2.1760  
1                 2   rt_0        6.97                 2.2090  
2                 2   rt_0        4.43                 2.1350  
3                 1   rt_0        2.40                 2.1050  
4                 1   rt_0        2.12                 2.2220  
...             ...    ...         ...                    ...  
2520              1  rt_12        2.99                 2.4900  
2521              1  rt_18        2.71                 2.2959  
2522              1   rt_0        2.06                 2.0900  
2523              2   rt_0        1.95                 2.1400  
2524              1   rt_0        1.97                 2.0300  

[2525 rows x 9 columns]

In [25]:
# Filtrar columnas relevantes para el análisis
selected_columns = [
    "fabricaAlimento", "granja_anonym", "genética", 
    "pesoInicial", "pesoFinal", "semesreSalida", 
    "asesor", "mortalidad", "conversionAlimenticia"
]

data_filtered = df[selected_columns]

# Generar análisis descriptivo
numeric_summary = data_filtered.describe(include=[float, int])
categorical_summary = data_filtered.describe(include=[object])

# Verificar valores faltantes
missing_values_summary = data_filtered.isnull().sum()

# Exportar reportes


In [26]:
missing_values_summary

fabricaAlimento          0
granja_anonym            0
genética                 0
pesoInicial              0
pesoFinal                0
semesreSalida            0
asesor                   0
mortalidad               0
conversionAlimenticia    0
dtype: int64

In [27]:
data_encoded = pd.get_dummies(data_filtered, columns=[
    "fabricaAlimento", "granja_anonym", "genética", "semesreSalida", "asesor"
], drop_first=True)

# Separar predictores y objetivo
X = data_encoded.drop(columns=["conversionAlimenticia"])
y = data_encoded["conversionAlimenticia"]

In [28]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.tree import DecisionTreeRegressor
from sklearn.ensemble import GradientBoostingRegressor
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error



In [29]:
# Dividir en conjunto de entrenamiento y prueba
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Entrenar y evaluar modelos
results = {}

# 1. Regresión Lineal
lr_model = LinearRegression()
lr_model.fit(X_train, y_train)
y_pred_lr = lr_model.predict(X_test)
results['Linear Regression'] = {
    "R2": r2_score(y_test, y_pred_lr),
    "MAE": mean_absolute_error(y_test, y_pred_lr),
    "RMSE": mean_squared_error(y_test, y_pred_lr, squared=False)
}

# 2. Árbol de Decisión
dt_model = DecisionTreeRegressor(random_state=42)
dt_model.fit(X_train, y_train)
y_pred_dt = dt_model.predict(X_test)
results['Decision Tree'] = {
    "R2": r2_score(y_test, y_pred_dt),
    "MAE": mean_absolute_error(y_test, y_pred_dt),
    "RMSE": mean_squared_error(y_test, y_pred_dt, squared=False)
}

# 3. Gradient Boosting
gb_model = GradientBoostingRegressor(random_state=42)
gb_model.fit(X_train, y_train)
y_pred_gb = gb_model.predict(X_test)
results['Gradient Boosting'] = {
    "R2": r2_score(y_test, y_pred_gb),
    "MAE": mean_absolute_error(y_test, y_pred_gb),
    "RMSE": mean_squared_error(y_test, y_pred_gb, squared=False)
}



In [30]:
# Convertir resultados a DataFrame para mejor visualización
pd.DataFrame(results).T

R2       MAE      RMSE
Linear Regression  0.398822  0.084539  0.116976
Decision Tree     -0.036215  0.112706  0.153575
Gradient Boosting  0.419803  0.085309  0.114916

In [32]:
import matplotlib.pyplot as plt

# Obtener las importancias de las características del modelo Gradient Boosting
feature_importances = pd.Series(gb_model.feature_importances_, index=X.columns)
feature_importances_sorted = feature_importances.sort_values(ascending=False)

# Visualizar las importancias
plt.figure(figsize=(10, 6))
feature_importances_sorted.head(10).plot(kind='bar')
plt.title('Top 10 Variables Más Importantes (Gradient Boosting)')
plt.ylabel('Importancia')
plt.xlabel('Variable')
plt.tight_layout()
plt.show()

# Mostrar las importancias en un DataFrame
important_features_df = feature_importances_sorted.reset_index()
important_features_df.columns = ['Variable', 'Importancia']



C:\Users\Asus\AppData\Local\Temp\ipykernel_6360\1496521656.py:14: UserWarning: Matplotlib is currently using module://matplotlib_inline.backend_inline, which is a non-GUI backend, so cannot show the figure.
  plt.show()


In [34]:
import matplotlib.pyplot as plt

# Obtener las importancias de las características del modelo Gradient Boosting
feature_importances = pd.Series(gb_model.feature_importances_, index=X.columns)
feature_importances_sorted = feature_importances.sort_values(ascending=False)

# Visualizar las importancias
plt.figure(figsize=(10, 6))
feature_importances_sorted.head(10).plot(kind='bar')
plt.title('Top 10 Variables Más Importantes (Gradient Boosting)')
plt.ylabel('Importancia')
plt.xlabel('Variable')
plt.tight_layout()
plt.show()





C:\Users\Asus\AppData\Local\Temp\ipykernel_6360\1669027260.py:14: UserWarning: Matplotlib is currently using module://matplotlib_inline.backend_inline, which is a non-GUI backend, so cannot show the figure.
  plt.show()


In [35]:
# Mostrar las importancias en un DataFrame
important_features_df = feature_importances_sorted.reset_index()
important_features_df.columns = ['Variable', 'Importancia']
important_features_df

Variable  Importancia
0              mortalidad     0.147997
1             pesoInicial     0.117017
2               pesoFinal     0.093945
3    fabricaAlimento_pp_3     0.081590
4             asesor_rt_2     0.072888
..                    ...          ...
124    granja_anonym_gr_5     0.000000
125   granja_anonym_gr_49     0.000000
126   granja_anonym_gr_46     0.000000
127   granja_anonym_gr_45     0.000000
128           asesor_rt_9     0.000000

[129 rows x 2 columns]

In [36]:
# Calcular los residuales
residuals = y_test - y_pred_gb

# Crear un DataFrame con las predicciones, valores reales y residuales
deviation_analysis = pd.DataFrame({
    "Real": y_test,
    "Predicted": y_pred_gb,
    "Residual": residuals
}, index=y_test.index)

# Agregar las características originales para un análisis más profundo
deviation_analysis = pd.concat([deviation_analysis, X_test.reset_index(drop=True)], axis=1)

# Identificar los lotes con mayor desviación (residuales más altos o más bajos)
most_deviated = deviation_analysis.sort_values(by="Residual", ascending=False).head(10)
least_deviated = deviation_analysis.sort_values(by="Residual", ascending=True).head(10)




In [37]:
least_deviated

Real  Predicted  Residual  pesoInicial  pesoFinal  mortalidad  \
1330  1.8376   2.324775 -0.487175          NaN        NaN         NaN   
2370  1.9500   2.412207 -0.462207          NaN        NaN         NaN   
1655  1.8853   2.286994 -0.401694          NaN        NaN         NaN   
2510  2.0910   2.485230 -0.394230          NaN        NaN         NaN   
1097  1.9215   2.247519 -0.326019          NaN        NaN         NaN   
1607  2.0881   2.406668 -0.318568          NaN        NaN         NaN   
2283  1.8300   2.145319 -0.315319          NaN        NaN         NaN   
278   2.1202   2.417085 -0.296885      30.2539   124.2860        1.21   
1608  2.1355   2.431404 -0.295904          NaN        NaN         NaN   
69    1.8370   2.108928 -0.271928      28.9975    88.3013        1.74   

     fabricaAlimento_pp_1 fabricaAlimento_pp_2 fabricaAlimento_pp_3  \
1330                  NaN                  NaN                  NaN   
2370                  NaN                  NaN                  NaN   
1655                  NaN                  NaN                  NaN   
2510                  NaN                  NaN                  NaN   
1097                  NaN                  NaN                  NaN   
1607                  NaN                  NaN                  NaN   
2283                  NaN                  NaN                  NaN   
278                 False                False                False   
1608                  NaN                  NaN                  NaN   
69                  False                False                False   

     fabricaAlimento_pp_4  ... asesor_rt_23 asesor_rt_24 asesor_rt_25  \
1330                  NaN  ...          NaN          NaN          NaN   
2370                  NaN  ...          NaN          NaN          NaN   
1655                  NaN  ...          NaN          NaN          NaN   
2510                  NaN  ...          NaN          NaN          NaN   
1097                  NaN  ...          NaN          NaN          NaN   
1607                  NaN  ...          NaN          NaN          NaN   
2283                  NaN  ...          NaN          NaN          NaN   
278                 False  ...        False        False        False   
1608                  NaN  ...          NaN          NaN          NaN   
69                  False  ...        False        False        False   

     asesor_rt_3 asesor_rt_4 asesor_rt_5 asesor_rt_6 asesor_rt_7 asesor_rt_8  \
1330         NaN         NaN         NaN         NaN         NaN         NaN   
2370         NaN         NaN         NaN         NaN         NaN         NaN   
1655         NaN         NaN         NaN         NaN         NaN         NaN   
2510         NaN         NaN         NaN         NaN         NaN         NaN   
1097         NaN         NaN         NaN         NaN         NaN         NaN   
1607         NaN         NaN         NaN         NaN         NaN         NaN   
2283         NaN         NaN         NaN         NaN         NaN         NaN   
278        False       False       False       False       False       False   
1608         NaN         NaN         NaN         NaN         NaN         NaN   
69         False       False       False       False       False       False   

     asesor_rt_9  
1330         NaN  
2370         NaN  
1655         NaN  
2510         NaN  
1097         NaN  
1607         NaN  
2283         NaN  
278        False  
1608         NaN  
69         False  

[10 rows x 132 columns]

In [38]:
# Variables categóricas a analizar
categorical_vars = ["asesor", "fabricaAlimento", "granja_anonym", "genética"]

# Crear un DataFrame para guardar los resultados
marginal_effects = []

# Calcular promedios, mejores y peores niveles
for var in categorical_vars:
    category_stats = data_filtered.groupby(var)["conversionAlimenticia"].mean()
    best_level = category_stats.idxmin()
    worst_level = category_stats.idxmax()
    best_value = category_stats.min()
    worst_value = category_stats.max()
    marginal_effect = worst_value - best_value

    # Guardar resultados
    marginal_effects.append({
        "Variable": var,
        "Best Level": best_level,
        "Worst Level": worst_level,
        "Best Conversion": best_value,
        "Worst Conversion": worst_value,
        "Marginal Effect": marginal_effect
    })

# Convertir a DataFrame para visualización
marginal_effects_df = pd.DataFrame(marginal_effects)

# Calcular la diferencia en alimento y su valorización
# Suposición: 190 kg de peso ganado por cada cerdo (220 kg - 30 kg)
population = 200_000
weight_gain = 190
cost_per_kg = 2500

marginal_effects_df["Feed Difference (kg)"] = marginal_effects_df["Marginal Effect"] * weight_gain * population
marginal_effects_df["Valuation (COP)"] = marginal_effects_df["Feed Difference (kg)"] * cost_per_kg


In [39]:
marginal_effects_df

Variable Best Level Worst Level  Best Conversion  Worst Conversion  \
0           asesor       rt_2       rt_11         2.069838          2.427260   
1  fabricaAlimento       pp_0        pp_6         2.146760          2.377620   
2    granja_anonym       gr_2       gr_25         2.064328          2.621629   
3         genética       lg_5        lg_6         2.214083          2.530129   

   Marginal Effect  Feed Difference (kg)  Valuation (COP)  
0         0.357422          1.358205e+07     3.395511e+10  
1         0.230860          8.772681e+06     2.193170e+10  
2         0.557301          2.117743e+07     5.294358e+10  
3         0.316046          1.200975e+07     3.002437e+10

In [40]:
descriptive_stats = df.describe(include='all')

In [41]:
descriptive_stats

loteNo departamento_anonym empresa_anonym  \
count                       2525                2525           2525   
unique                      2525                   8             77   
top     5e4604c71c555bfa922bd644                dp_3          em_46   
freq                           1                1215            250   
mean                         NaN                 NaN            NaN   
std                          NaN                 NaN            NaN   
min                          NaN                 NaN            NaN   
25%                          NaN                 NaN            NaN   
50%                          NaN                 NaN            NaN   
75%                          NaN                 NaN            NaN   
max                          NaN                 NaN            NaN   

       granja_anonym proveedorCorregido_anonym fabricaAlimento genética  \
count           2525                      2525            2525     2525   
unique            87                        11               9        7   
top            gr_48                      pn_0            pp_8     lg_0   
freq             250                      1100             664     2114   
mean             NaN                       NaN             NaN      NaN   
std              NaN                       NaN             NaN      NaN   
min              NaN                       NaN             NaN      NaN   
25%              NaN                       NaN             NaN      NaN   
50%              NaN                       NaN             NaN      NaN   
75%              NaN                       NaN             NaN      NaN   
max              NaN                       NaN             NaN      NaN   

        duracionCeba    edadFinal  edadInicial  ...    pesoFinal  fechaSalida  \
count    2525.000000  2525.000000  2525.000000  ...  2525.000000         2525   
unique           NaN          NaN          NaN  ...          NaN          685   
top              NaN          NaN          NaN  ...          NaN    1/02/2021   
freq             NaN          NaN          NaN  ...          NaN           45   
mean       84.353299   157.153844    72.800545  ...   113.904912          NaN   
std        10.956194    10.141550     5.659016  ...     9.841921          NaN   
min        19.600000    92.600000    53.000000  ...    81.560000          NaN   
25%        77.000000   151.000000    70.000000  ...   107.589700          NaN   
50%        83.313100   156.450000    71.000000  ...   114.886700          NaN   
75%        91.100000   163.000000    74.000000  ...   120.890000          NaN   
max       144.000000   206.000000   104.820000  ...   138.755000          NaN   

         anioSalida  semesreSalida  anioSemestreSalida asesor  \
count   2525.000000    2525.000000                2525   2525   
unique          NaN            NaN                   9     26   
top             NaN            NaN             2021-S1  rt_18   
freq            NaN            NaN                 375    306   
mean    2020.756436       1.472871                 NaN    NaN   
std        1.105240       0.499362                 NaN    NaN   
min     2019.000000       1.000000                 NaN    NaN   
25%     2020.000000       1.000000                 NaN    NaN   
50%     2021.000000       1.000000                 NaN    NaN   
75%     2022.000000       2.000000                 NaN    NaN   
max     2023.000000       2.000000                 NaN    NaN   

       numFinalAnimales  numInicialAnimales   mortalidad  \
count       2525.000000         2525.000000  2525.000000   
unique              NaN                 NaN          NaN   
top                 NaN                 NaN          NaN   
freq                NaN                 NaN          NaN   
mean         588.589307          601.573465     1.880392   
std          704.765578          720.830793     1.633003   
min           13.000000           13.000000     0.000000   
25%          163.000000          167.0

In [44]:
cols = [
     'departamento_anonym', 'empresa_anonym', 'granja_anonym',
       'proveedorCorregido_anonym', 'fabricaAlimento', 'genética',
       'duracionCeba', 'edadFinal', 'edadInicial', 'pesoInicial', 'pesoFinal',
       'fechaSalida', 'anioSalida', 'semesreSalida', 'anioSemestreSalida',
       'asesor', 'numFinalAnimales', 'numInicialAnimales', 'mortalidad',
       'conversionAlimenticia'
    
]

In [45]:
correlation_matrix = df[cols].corr()

ValueError: could not convert string to float: 'dp_0'

In [ ]:
# Identificar columnas con alta correlación o posibles redundancias (correlación mayor a 0.8 o menor a -0.8)
high_correlation = correlation_matrix[(correlation_matrix > 0.8) | (correlation_matrix < -0.8)]
high_correlation = high_correlation.where(pd.notnull(high_correlation), '')